### ISSUE TITLE
Incorrect route calculation, Norway, Vossevangen.

### ISSUE THREAD

**Issue by PaulStets** [osmandapp/Osmand#3978]
OsmAnd does not use this road for car navigation: https://www.openstreetmap.org/way/28212180#map=18/60.63062/6.41574
![19532567_1874800052766800_560386715_o](https://user-images.githubusercontent.com/20681525/27584334-46718ca0-5b40-11e7-9c62-1cde8c794d1c.png)

**Comment by Gazer75** [comment]
Strange that it is actually working properly in that picture.
Try this one: https://www.openstreetmap.org/directions?engine=mapzen_car&route=60.6287%2C6.3929%3B60.6168%2C6.3684#map=15/60.6228/6.3853

Doing that route with latest map data and OsmAnd version on my Galaxy Note will route through https://www.openstreetmap.org/way/169702172

**Comment by PaulStets** [comment]
Here is the same route in OSM: https://www.openstreetmap.org/directions?engine=mapzen_car&route=60.6333%2C6.4192%3B60.6298%2C6.4103#map=17/60.63091/6.41476

**Comment by Gazer75** [comment]
Yep and it routed like your image when I tested it now. But my route example above still routes trough Burekkje.

Looked this first example with my dad on his phone yesterday and he might be using older version of OsmAand.

**Comment by vshcherb** [comment]
This a known mechanism for OsmAnd sometimes it prefers to go to a higher class road. It is configurable (for example bicycle routing doesn't do it) but it has disadvantages in other examples

**Comment by Gazer75** [comment]
Think you misunderstand. The picture example above is actually correct routing. Lekvebakken is not open for through traffic using motorized vehicles.
It should be impossible for OsmAnd in car mode to route through a road tagged with motor_vehicles=destination, but it still happen. 
It causes som serious problems for people because they can end up at a road that is not legal to drive through, and have a very hard time turning around.

**Comment by vshcherb** [comment]
Yes destination should be used only for destination, this is a valid issue. Today we only deprioritize roads due to technical limitations.
